src/cost_estimates/tests/fee_medians.rs

```diff
@@ -0,0 +1,391 @@
+use std::{env, path::PathBuf};
+use time::Instant;
+
+use rand::seq::SliceRandom;
+use rand::Rng;
+
+use cost_estimates::metrics::CostMetric;
+use cost_estimates::{EstimatorError, FeeEstimator};
+use vm::costs::ExecutionCost;
+
+use chainstate::burn::ConsensusHash;
+use chainstate::stacks::db::{StacksEpochReceipt, StacksHeaderInfo};
+use chainstate::stacks::events::StacksTransactionReceipt;
+use types::chainstate::{BlockHeaderHash, BurnchainHeaderHash, StacksBlockHeader, StacksWorkScore};
+use types::proof::TrieHash;
+use util::hash::{to_hex, Hash160, Sha512Trunc256Sum};
+use util::vrf::VRFProof;
+
+use crate::chainstate::stacks::{
+    CoinbasePayload, StacksTransaction, TokenTransferMemo, TransactionAuth,
+    TransactionContractCall, TransactionPayload, TransactionSpendingCondition, TransactionVersion,
+};
+use crate::core::StacksEpochId;
+use crate::cost_estimates::fee_medians::WeightedMedianFeeRateEstimator;
+use crate::cost_estimates::metrics::ProportionalDotProduct;
+use crate::cost_estimates::FeeRateEstimate;
+use crate::types::chainstate::StacksAddress;
+use crate::vm::types::{PrincipalData, StandardPrincipalData};
+use crate::vm::Value;
+use cost_estimates::fee_medians::fee_rate_estimate_from_sorted_weighted_fees;
+use cost_estimates::fee_medians::FeeRateAndWeight;
+use cost_estimates::tests::common::*;
+
+/// Tolerance for approximate comparison.
+const error_epsilon: f64 = 0.1;
```

Comment:
If we're comparing fee rates, then won't this be a somewhat big epsilon for when blocks aren't full?  i.e. the fee rate in this case would be small.

Reply:
I was being kind of lazy in not just making this a reusable util function.

We should make a reusable `is_close` for floats, and make epsilon to the base value, i.e., `abs(a-b)/a < eps`... then make it 0.001 or something.

Then, this derivative is_close can use the float one.

Reply:
The code itself is fine for now; I was only commenting on the `epsilon` value.